### PR TITLE
📖 Improve formatting on the docs website

### PR DESCRIPTION
**Pull request by m1kola**
# Description

Fixes a number of formatting/rendering issues. Mostly around lists, admonitions (notes, warnings, etc) and collapsible text.

> [!TIP]
> Recommend reviewing the changes with `git diff --ignore-all-space` (`-w`) option. GitHub UI also has this option.
>  Screenshot 2024-11-12 at 15 59 15

> [!TIP]
> We also have docs deployment preview. See the comment from netlify bot below.

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: docs/tutorials/downgrade-extension.md
Comment (m1kola) [inline docs/tutorials/downgrade-extension.md:56]: This is the only content change because it was very odd to read this.

Everything else is just formatting.

File: docs/concepts/upgrade-support.md
Comment (m1kola) [inline docs/concepts/upgrade-support.md:67]: `!!! note` This renders well in our mkdocs website, but not on GitHub. `[!NOTE]` renders well on GitHub, but not in mkdocs.

Ref: https://squidfunk.github.io/mkdocs-material/reference/admonitions/#classic-admonitions-docsstylesheetsextracss

We are changing these admonitions in `docs` dir because these are docs which will be rendered on the website. Leaving everything outside of docs dir as is - these will likely be viewed via GitHub UI.

File: CONTRIBUTING.md
Comment (m1kola) [inline CONTRIBUTING.md:47]: It was rendering fine in GH UI. However this doc is also used in the website. This formatting works fine in both GH and on the website.

File: docs/concepts/upgrade-support.md
Comment (trgeiger) [inline docs/concepts/upgrade-support.md:63]: I'm having trouble commenting on the correct line for some reason on Github, but the bullet-pointed list on lines 20-23 looks like it's intended to have multiple levels of indentation. I think the first and fourth bullets can just be removed and the middle two un-indented.

Comment (trgeiger) [reply docs/concepts/upgrade-support.md:63]: Also, line 54 can be changed to "Setting the 'upgradeConstraintPolicy' to:"

Comment (m1kola) [reply docs/concepts/upgrade-support.md:65]: @trgeiger good catch!

Is [this](https://github.com/operator-framework/operator-controller/compare/77c1ae7943cc4d0ff0de75ccde571a442ff5d4d2..859f2ba7a64cb928332e3506746dc87856b3d7c5) what you had in mind?

**Review by camilamacedo86 — APPROVED** [review]
Assuming that you finished (since I cannot see in the title WIP)

Hi @m1kola 

The emoji here is 📖 not 🌱 

🌱 is only for things that does not bring impacted to the end users at all (i.e changes in the e2e tests)

So could you please fix the emoji?

Otherwise, 

I gave a look on this one with the preview and all shows fine 🥇 
Great work !!! Well done

/lgtm

**Comment by m1kola** [comment]
Forgot we have a separate emoji for docs. Updated.

**Review by trgeiger — CHANGES_REQUESTED** [review]
Requesting a few small changes

File: docs/howto/derive-service-account.md
Comment (m1kola) [reply docs/howto/derive-service-account.md:1]: Nice one! That's much better. I'll push changes soon.

Also noticed that the list is not rendering as a list underneath "Putting it all together"

**Review by trgeiger — APPROVED** [review]

**Comment by m1kola** [comment]
Had to rebase due to https://github.com/operator-framework/operator-controller/pull/1228

**Comment by m1kola** [comment]
Blocked by https://github.com/operator-framework/operator-controller/pull/1451. Will require another rebase once it merges.

**Review by camilamacedo86 — COMMENTED** [review]
/lgtm
/approve

**Comment by m1kola** [comment]
Rebased on top of https://github.com/operator-framework/operator-controller/pull/1451 so it can pass `upgrade-e2e`.

**Pull request merged**
